src/main/java/com/upserve/uppend/lookup/LookupMetadata.java

```diff
@@ -101,26 +117,32 @@ public void writeTo(Path path) throws IOException {
         log.trace("writing metadata to path: {}", path);
         Path tmpPath = path.resolveSibling(path.getFileName() + ".tmp");
         try (FileChannel chan = FileChannel.open(tmpPath, StandardOpenOption.READ, StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
-            try (DataOutputStream dout = new DataOutputStream(new BufferedOutputStream(Channels.newOutputStream(chan), 1024))) {
-                if (keyLength != minKey.byteLength()) {
-                    throw new IllegalStateException("expected key length (" + keyLength + ") does not agree with minKey length (" + minKey.byteLength() + ")");
-                }
-                if (keyLength != maxKey.byteLength() || keyLength != maxKey.byteLength()) {
-                    throw new IllegalStateException("expected key length (" + keyLength + ") does not agree with maxKey length (" + maxKey.byteLength() + ")");
-                }
+            if (keyLength != minKey.byteLength()) {
+                throw new IllegalStateException("expected key length (" + keyLength + ") does not agree with minKey length (" + minKey.byteLength() + ")");
+            }
+            if (keyLength != maxKey.byteLength() || keyLength != maxKey.byteLength()) {
+                throw new IllegalStateException("expected key length (" + keyLength + ") does not agree with maxKey length (" + maxKey.byteLength() + ")");
+            }
 
-                dout.writeInt(keyLength);
-                dout.writeInt(numKeys);
-                dout.write(minKey.bytes());
-                dout.write(maxKey.bytes());
+            IntegratedIntCompressor iic = new IntegratedIntCompressor();
+            int[] compressedKeyStorageOrder = iic.compress(keyStorageOrder);
+            int numKeysCompressed = compressedKeyStorageOrder.length;
```

Comment:
I think these names could be improved. In particular, `numKeysCompressed` sounds like it's a compressed version of `numKeys`.

Reply:
d1d8040